File: 0388-longest-absolute-file-path/0388-longest-absolute-file-path.java
class Solution {
    public int lengthLongestPath(String input) {
        // 1. Instantly split the entire string into clean, individual lines
        String[] lines = input.split("\n");
        
        Stack<Integer> stack = new Stack<>();
        int maxLen = 0;
        
        for (String line : lines) {
            /* 
             * 2. FIND THE DEPTH LEVEL:
             * Look for the last occurrence of the tab character '\t'.
             * If there are no tabs, lastIndexOf returns -1.
             * Adding 1 gives us the perfect depth level (0 for root, 1 for subfolder, etc.)
             */
            int depth = line.lastIndexOf("\t") + 1;
            
            // 3. ALIGN THE STACK: Move back to the correct parent folder level
            while (stack.size() > depth) {
                stack.pop();
            }
            
            // 4. CALCULATE CLEAN NAME LENGTH: Total length minus the tab characters
            int nameLen = line.length() - depth;
            
            // 5. COMPUTE CUMULATIVE PATH LENGTH:
            // If the stack is empty, we are at the root level.
            // Otherwise, add the parent folder length plus 1 for the '/' character.
            int parentLen = stack.isEmpty() ? 0 : stack.peek() + 1;
            int currLen = parentLen + nameLen;
            
            // 6. CHECK FILE VS DIRECTORY:
            if (line.contains(".")) {
                // It has a dot, so it's a file! Update our maximum tracker.
                maxLen = Math.max(maxLen, currLen);
            } else {
                // It's a directory! Push its total path length onto the stack.
                stack.push(currLen);
            }
        }
        
        return maxLen;
    }
}

File: 0388-longest-absolute-file-path/0388-longest-absolute-file-path.cpp
class Solution {
public:
    int lengthLongestPath(string input) {
        stringstream ss(input);
        string line;
        stack<int> st; // Stores cumulative length of parent directories
        int maxlen = 0;
        
        while (getline(ss, line, '\n')) {
            // 1. Calculate depth by counting consecutive tab characters at the front
            int depth = 0;
            while (depth < line.length() && line[depth] == '\t') {
                depth++;
            }
            
            // 2. Adjust stack position to align with current tree depth level
            while (st.size() > depth) {
                st.pop();
            }
            
            // 3. Extract the clean token name (strip away the front tabs)
            string name = line.substr(depth);
            
            // 4. Compute cumulative length from parent directory
            int parent_len = st.empty() ? 0 : st.top();
            int curr_len = parent_len + name.length();
            
            // 5. If it contains a dot, it's a file -> update our global maximum
            if (name.find('.') != string::npos) {
                maxlen = max(maxlen, curr_len);
            } 
            // 6. If it's a directory, push it onto the stack (+1 accounts for the future '/' separator)
            else {
                st.push(curr_len + 1);
            }
        }
        
        return maxlen;
    }
};